### PR TITLE
[DE-340] refactor: move CI/CD long parts in scripts

### PR DESCRIPTION
**Pull request by lmontier-pass**
# DE PR

## Describe your changes

* Move hard CI parts in scripts
* Fix find_intersections.sh

## Jira ticket number and/or notion link

https://www.notion.so/passcultureapp/880351701699470cb129e4e0ad6bbcba?v=bfba95ea66224a80aed7b5291454389a&p=3d52fd8ba43b4115aad3a5ecace092ff&pm=s

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [x] ⏰ no, but I created a ticket

### CODE REVIEW

File: .github/workflows/scripts/run_dbt_changed_models.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Check if TARGET_BRANCH is provided
+if [ -z "$1" ]; then
+  echo "TARGET_BRANCH is required as the first argument"
+  exit 1
+fi
+
+TARGET_BRANCH=$1
+children="+1"
```
Comment (valoumiaou) [inline .github/workflows/scripts/run_dbt_changed_models.sh:10]: est ce qu'on le mettrait pas comme 2eme argument avec valeur par default +1 ? il se peut qu'on veuille tester plus loin dans le futur ou bien o, verra a ce moment la

File: .github/workflows/reusable_dbt_compile.yml [hunk 2]
```diff
@@ -85,36 +83,26 @@ jobs:
           ENV_SHORT_NAME: ${{ inputs.ENV_SHORT_NAME }}
         run: dbt compile --profile CI --target $ENV_SHORT_NAME --vars "{'ENV_SHORT_NAME':'$ENV_SHORT_NAME'}" 
       - name: "Run modified models & 1st order childrens"
-        working-directory: "./orchestration/dags/data_gcp_dbt/"
+        working-directory: .
```
Comment (valoumiaou) [inline .github/workflows/reusable_dbt_compile.yml:86]: c'est caché dans le script, pas cohérent avec le reste, risque de créer des oublis on a envie de bien spectifier qu'on est dans le folder dbt pour runner des commande dbt no ?

File: .github/workflows/scripts/run_dbt_changed_models.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+WORKING_DIR="orchestration/dags/data_gcp_dbt/"
```
Comment (valoumiaou) [inline .github/workflows/scripts/run_dbt_changed_models.sh:11]: caché cf plus haut

File: .github/workflows/scripts/run_dbt_changed_models.sh (hunk 1 above)
Comment (lmontier-pass) [reply .github/workflows/scripts/run_dbt_changed_models.sh:10]: Je pense que quand on voudra changer ça dans le futur si ce besoin apparaît, on pourra alors le paramétriser très facilement. Mais vu qu'ici le paramètre ne bouge pas, je trouve que ça ne sert pas à grand chose. 
Ok pour toi?

**Pull request merged**